chromium_src/ui/views/widget/widget_delegate.h

```diff
@@ -0,0 +1,32 @@
+/* Copyright (c) 2024 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_CHROMIUM_SRC_UI_VIEWS_WIDGET_WIDGET_DELEGATE_H_
+#define BRAVE_CHROMIUM_SRC_UI_VIEWS_WIDGET_WIDGET_DELEGATE_H_
+
+#define set_desired_bounds_delegate                                 \
+  set_desired_bounds_delegate_unused();                             \
+  gfx::Point get_desired_position() {                               \
+    return desired_position_delegate_.Run();                        \
+  }                                                                 \
+  bool has_desired_position_delegate() const {                      \
+    return static_cast<bool>(desired_position_delegate_);           \
+  }                                                                 \
+  void set_desired_position_delegate(                               \
+      base::RepeatingCallback<gfx::Point()> delegate) {             \
+    desired_position_delegate_ = std::move(delegate);               \
+  }                                                                 \
+                                                                    \
+ private:                                                           \
+  base::RepeatingCallback<gfx::Point()> desired_position_delegate_; \
```

Comment:
can you add some comments why this new callback is introduced instead of using `params_.desired_bounds_delegate`?

Reply:
Sure. Done.